src/WebformBuilder.js

```diff
@@ -1132,7 +1132,12 @@ export default class WebformBuilder extends Component {
         parent.formioComponent.removeChildComponent(component);
       }
       if (component.input && componentInstance && componentInstance.parent) {
-        _.unset(componentInstance._data, componentInstance.key);
+        if (Array.isArray(parent.formioComponent.component.defaultValue)) {
+          parent.formioComponent.component.defaultValue.forEach(v => _.unset(v, componentInstance.key));
+        }
+        else if (typeof parent.formioComponent.component.defaultValue === 'object') {
+          _.unset(parent.formioComponent.component.defaultValue, componentInstance.key);
+        }
```

Comment:
I believe we should have more reliable check for parent.formioComponent.component.defaultValue. What if there is no parent or  parent.formioComponent. You can use lodash to get the default value: _.get(parent, 'formioComponent.component.defaultValue')

Reply:
There is a check for 'parent' at the very top of the method, so I added check for parent.formioComponent and used _.get to get it's default value